compiler/rustc_middle/src/ty/util.rs

```diff
@@ -591,7 +590,8 @@ impl<'tcx> TyCtxt<'tcx> {
 
     /// Returns `true` if this is a `static` item with the `#[thread_local]` attribute.
     pub fn is_thread_local_static(self, def_id: DefId) -> bool {
-        self.codegen_fn_attrs(def_id).flags.contains(CodegenFnAttrFlags::THREAD_LOCAL)
+        matches!(self.def_kind(def_id), DefKind::Static(_))
+            && self.has_attr(def_id, sym::thread_local)
```

Comment:
`codegen_fn_attrs` do many things. For example, if we use `is_thread_local_static` for non-fn items with `track_caller` attribute, rustc can cause an ICE in the following lines. I think we need to stop using `codegen_fn_attrs` to avoid things like this.

Given the following code:

```rust
#[track_caller]
static FOO: usize = 3;

fn main() {}
```

rustc causes an ICE because of `is_thread_local_static`.

https://github.com/rust-lang/rust/blob/2176e3a7a4a8dfbea92f3104244fbf8fad4faf9a/compiler/rustc_hir_analysis/src/collect.rs#L1741-L1742

Since `is_thread_local_static` has not checked if def_kind is static before, we can probably remove `matches!(self.def_kind(def_id), DefKind::Static(_))`

Reply:
`codegen_fn_attrs` will still be called on statics by codegen backends even after this PR.